### PR TITLE
type/compatibility: fix show create table output of generated column

### PR DESCRIPTION
**Pull request by pingyu**
<!--
Thank you for contributing to TiDB! Please read TiDB's [CONTRIBUTING](https://github.com/pingcap/tidb/blob/master/CONTRIBUTING.md) document **BEFORE** filing this PR.
-->

### What problem does this PR solve? <!--add issue link with summary if exists-->
fix incompatible output of `show create table` for generated column

### What is changed and how it works?
Change method for restoring `ast.ColumnOptionGenerated` from `Format` to `Restore`
Change related `cast` expression in parser
Related issue #9457

### Check List <!--REMOVE the items that are not applicable-->

Tests <!-- At least one of them must be included. -->

 - Unit test
 - Manual test (add detailed scripts or steps below)
```sql
mysql> 
mysql> CREATE TABLE t20( first_name varchar(10), last_name varchar(10), full_name varchar(255) AS (CONCAT(first_name,' ' ,last_name) ));
Query OK, 0 rows affected (0.01 sec)

mysql> show create table t20;
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                                                                                                                      |
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| t20   | CREATE TABLE `t20` (
  `first_name` varchar(10) DEFAULT NULL,
  `last_name` varchar(10) DEFAULT NULL,
  `full_name` varchar(255) GENERATED ALWAYS AS (concat(`first_name`, ' ', `last_name`)) VIRTUAL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> select tidb_version();
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                                                                                                                                                                                                                                |
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v3.0.0-beta-170-ga64f3092f-dirty
Git Commit Hash: a64f3092f49fff6ab0538dc3bdaf43cb53553a3f
Git Branch: show_create_table_virtual
UTC Build Time: 2019-03-07 04:05:05
GoVersion: go version go1.11.2 darwin/amd64
Race Enabled: false
TiKV Min Version: 2.1.0-alpha.1-ff3dd160846b7d1aed9079c389fc188f7f5ea13e
Check Table Before Drop: false |
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> 
```

Code changes

 - Has exported function/method change
No.
 - Has exported variable/fields change
No.
 - Has interface methods change
No.
 - Has persistent data change
No.

Side effects

 - Possible performance regression
No.
 - Increased code complexity
No.
 - Breaking backward compatibility
No.

Related changes

 - Need to cherry-pick to the release branch
No.
 - Need to update the documentation
No.
 - Need to update the `tidb-ansible` repository
No.
 - Need to be included in the release note
No.

### CODE REVIEW

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -492,9 +498,12 @@ func columnDefToCol(ctx sessionctx.Context, offset int, colDef *ast.ColumnDef, o
 					return nil, nil, errors.Trace(err)
 				}
 			case ast.ColumnOptionGenerated:
-				var buf = bytes.NewBuffer([]byte{})
-				v.Expr.Format(buf)
```
Comment (zimulala) [inline ddl/ddl_api.go]: We do it here, then do we need to update https://github.com/pingcap/tidb/blob/master/ddl/ddl_api.go#L2395 too?

Comment (pingyu) [reply ddl/ddl_api.go]: @zimulala Review addressed. PTAL. Thanks.

```sql
mysql> CREATE TABLE t20( first_name varchar(10), last_name varchar(10), full_name varchar(255) AS (CONCAT(first_name,' ' ,last_name) ));
Query OK, 0 rows affected (0.00 sec)

mysql> show create table t20;
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                                                                                                                      |
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| t20   | CREATE TABLE `t20` (
  `first_name` varchar(10) DEFAULT NULL,
  `last_name` varchar(10) DEFAULT NULL,
  `full_name` varchar(255) GENERATED ALWAYS AS (concat(`first_name`, ' ', `last_name`)) VIRTUAL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

```

BEFORE:
```sql
mysql> ALTER TABLE t20 modify column full_name varchar(255) GENERATED ALWAYS AS (CONCAT(last_name,' ' ,first_name) ) VIRTUAL;
Query OK, 0 rows affected (0.01 sec)

mysql> show create table t20;
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                                                                                                                      |
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| t20   | CREATE TABLE `t20` (
  `first_name` varchar(10) DEFAULT NULL,
  `last_name` varchar(10) DEFAULT NULL,
  `full_name` varchar(255) GENERATED ALWAYS AS (concat(`last_name`, " ", `first_name`)) VIRTUAL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

```

AFTER:
```sql
mysql> 
mysql> ALTER TABLE t20 modify column full_name varchar(255) GENERATED ALWAYS AS (CONCAT(first_name,' ' ,last_name) ) VIRTUAL;
Query OK, 0 rows affected (0.01 sec)

mysql> show create table t20;
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                                                                                                                      |
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| t20   | CREATE TABLE `t20` (
  `first_name` varchar(10) DEFAULT NULL,
  `last_name` varchar(10) DEFAULT NULL,
  `full_name` varchar(255) GENERATED ALWAYS AS (concat(`first_name`, ' ', `last_name`)) VIRTUAL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> 
```

**Comment by zimulala** [comment]
Please fix CI, you can modify the code in /pingcap/tidb-test if needed.

**Comment by pingyu** [comment]
> Please fix CI, you can modify the code in [pingcap/tidb-test](https://github.com/pingcap/tidb-test) if needed.

OK, let me see see

**Pull request merged**